CONTRIBUTING.md

```diff
@@ -0,0 +1,104 @@
+  - Can you reliably reproduce the issue? And can you also reproduce it with older versions?
+
+#### How Do I Submit a Good Bug Report?
+
+> You must never report security related issues, vulnerabilities or bugs including sensitive information to the issue tracker, or elsewhere in public. Instead sensitive bugs must be sent by email to <dev@nimblehq.co>.
+
+We use GitHub issues to track bugs and errors. If you run into an issue with the project:
+
+- Open a [Bug Report's Issue](https://github.com/nimblehq/flutter-templates/issues/new?assignees=&labels=type+%3A+bug&projects=&template=bug_template.md&title=Fix%3A+).
+- Explain the behavior you would expect and the actual behavior.
+- Please provide as much context as possible and describe the *reproduction steps* that someone else can follow to recreate the issue on their own. This usually includes your code. For good bug reports you should isolate the problem and create a reduced test case.
+- Provide the information you collected in the previous section.
+
+Once it's filed:
+
+- The repository's maintainer will mark the issue as `triage` and try to reproduce the issue with your provided steps. If there are no reproduction steps or no obvious way to reproduce the issue, the team will ask you for further information. Bugs with the `triage` label will not be addressed until they are reproduced.
+- If the team is able to reproduce the issue, the `triage` label will be removed, and the issue will be left to be [implemented by someone](#your-first-code-contribution).
+
+### Your First Code Contribution
+
+If you're not sure where to start contributing to Flutter Templates, browse through the [Issues](https://github.com/nimblehq/flutter-templates/issues) and choose one that suits you best.
+
+### Pull Requests
+
+Please follow these steps to have your contribution considered by the maintainers:
+
+- Follow the styleguides
+- After you submit your pull request, verify that all [status checks](https://help.github.com/articles/about-status-checks/) are passing <details><summary>What if the status checks are failing?</summary>If a status check is failing, and you believe that the failure is unrelated to your change, please leave a comment on the pull request explaining why you believe the failure is unrelated. A maintainer will re-run the status check for you. If we conclude that the failure was a false positive, then we will open an issue to track that problem with our status check suite.</details>
+
+While the prerequisites above must be satisfied prior to having your pull request reviewed, the reviewer(s) may ask you to complete additional design work, tests, or other changes before your pull request can be ultimately accepted.
+
+## Styleguides
+### Git Branch Management
+
+Checkout our [Branch Management](https://nimblehq.co/compass/development/version-control/branch-management/) for more details about the branch formatting, naming, etc.
+
+### Git Commit Messages
```

Comment:
We can refer to our team convention here https://nimblehq.co/compass/development/code-conventions/flutter/ in a new section